packages/material-ui/src/Button/Button.test.js

```diff
@@ -367,4 +370,121 @@ describe('<Button />', () => {
+    };
+
+    it('should map the variant classkey to the component', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // see https://github.com/jsdom/jsdom/issues/2953
+        this.skip();
+      }
+
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+          ],
+        },
+      });
+
+      render(<WrappedComponent theme={theme} variant="test" />);
+
+      const style = window.getComputedStyle(screen.getByTestId('component'));
+      expect(style.getPropertyValue('background-color')).to.equal('rgb(255, 0, 0)');
+    });
+
+    it('should map the latest props combination classkey to the component', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // see https://github.com/jsdom/jsdom/issues/2953
+        this.skip();
+      }
+
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+            {
+              props: { variant: 'test', size: 'large' },
+              styles: { backgroundColor: 'rgb(0, 255, 0)' },
+            },
+          ],
+        },
+      });
+
+      render(<WrappedComponent theme={theme} variant="test" size="large" />);
+
+      const style = window.getComputedStyle(screen.getByTestId('component'));
+      expect(style.getPropertyValue('background-color')).to.equal('rgb(0, 255, 0)');
+    });
+
+    it('should not add classKey if all props are not a match', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // see https://github.com/jsdom/jsdom/issues/2953
+        this.skip();
+      }
+
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+            {
+              props: { variant: 'test', size: 'large' },
+              styles: { backgroundColor: 'rgb(0, 255, 0)' },
+            },
+          ],
+        },
+      });
+
+      render(<WrappedComponent theme={theme} size="large" />);
+
+      const style = window.getComputedStyle(screen.getByTestId('component'));
+      expect(style.getPropertyValue('background-color')).not.to.equal('rgb(0, 255, 0)');
+    });
+
+    it('should warn if the used variant is not defined in the theme', function test() {
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test1' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+          ],
+        },
+      });
+
+      expect(() => mount(<WrappedComponent theme={theme} variant="test" />)).toErrorDev([
+        [
```

Comment:
The error is thrown two times :\

Reply:
Thrown or logged in the console? The double console call is expected in react@16.13 since console.error is called every render (x2 in StrictMode).

Reply:
Now I see. Could you add a comment that this only happens in `StrictMode`? Then it's apparent that the behavior changes in `react@next`.

Reply:
Actually I am using just `console.error` directly, so `console.error` is called twice

Reply:
Done